packages/vuetify/src/components/VInfiniteScroll/VInfiniteScroll.tsx

```diff
@@ -0,0 +1,275 @@
+// Styles
+import './VInfiniteScroll.sass'
+
+// Components
+import { VBtn } from '@/components/VBtn'
+import { VProgressCircular } from '@/components/VProgressCircular'
+
+// Composables
+import { makeDimensionProps, useDimension } from '@/composables/dimensions'
+import { useIntersectionObserver } from '@/composables/intersectionObserver'
+import { useLocale } from '@/composables/locale'
+
+// Utilities
+import { computed, nextTick, onMounted, ref, watch } from 'vue'
+import { convertToUnit, defineComponent, genericComponent, useRender } from '@/util'
+
+// Types
+import type { PropType } from 'vue'
+
+export type InfiniteScrollSide = 'start' | 'end' | 'both'
+export type InfiniteScrollStatus = 'ok' | 'empty' | 'loading' | 'error'
+
+type InfiniteScrollSlot = {
+  side: InfiniteScrollSide
+  props: {
+    onClick: () => (side: InfiniteScrollSide) => void
+    color: string | undefined
+  }
+}
+
+type VInfiniteScrollSlots = {
+  default: [InfiniteScrollSlot]
+  loading: [InfiniteScrollSlot]
+  error: [InfiniteScrollSlot]
+  empty: [InfiniteScrollSlot]
+  'load-more': [InfiniteScrollSlot]
+}
+
+export const VInfiniteScrollIntersect = genericComponent()({
```

Comment:
@KaelWD since this doesn't have a slot should I leave it as **defineComponent**?